src/main/scala/scalang/Node.scala

```diff
@@ -71,6 +87,9 @@ object Node {
     } else {
       val cookie = randomCookie
       writeCookie(file, cookie)
+      val permissions = setAsJavaSet(Set(PosixFilePermission.OWNER_READ))
+      try
+        Files.setPosixFilePermissions(file.toPath, permissions)
```

Comment:
This won't work on Windows (right?) and the `try` just means it silently fails there? 

Files.setPosixFilePermissions also doesn't work on JDK < 8. Regrettably scalang (because of the version of Scala) only works properly (to the extent it does...) on JDK 6/7.

In summary, can we keep the PR to the smallest required change (the ability to do name and nodeconfig with a findOrGenerateCookie) in recognition of the legacy nature here?

Reply:
Sure, I can do that as well.  I intentionally put that change into a separate commit that I can simply remove.  I was hoping that I could improve on the existing functionality but I am not married to this proposal.